### PR TITLE
Automate the manual steps for rabbitmq

### PR DESCRIPTION
**Pull request by pkulkark**
This PR adds the manual steps for rabbitmq deployment into the ansible playbook. The changes are as follows:

- Set letsencrypt directory owner to rabbitmq

- Add vhost and user

- Download rabbitadmin executable

### CODE REVIEW

File: playbooks/roles/rabbitmq/tasks/main.yml [hunk 1]
```diff
@@ -155,20 +153,29 @@
...
+  rabbitmq_user:
+    user: "{{ RABBITMQ_USER }}"
+    password: "{{ RABBITMQ_PASS }}"
+    vhost: "{{ RABBITMQ_VHOST }}"
+    state: present
+
 - name: ensure vhost /test is present
   rabbitmq_vhost:
     name: /test
+    node: "rabbit@{{ RABBITMQ_HOSTNAME }}"
```
Comment (pkulkark) [inline playbooks/roles/rabbitmq/tasks/main.yml]: Note: This was added to correct the rabbitmqctl commands failing with "Hostname mismatched" errors.

File: playbooks/roles/rabbitmq/tasks/main.yml [hunk 2]
```diff
@@ -155,20 +153,29 @@
...
 
-- name: fetch the rabbitmqadmin script from the management webserver
-  get_url:
-    url: "https://{{ RABBITMQ_HOSTNAME }}:{{ RABBITMQ_HTTPS_PORT }}/cli/rabbitmqadmin"
-    dest: "{{ RABBITMQ_SCRIPTS_DIR }}/rabbitmqadmin"
-    owner: root
-    group: root
-    mode: 0700
+- name: download rabbitmqadmin executable
+  command: wget -P "{{ RABBITMQ_SCRIPTS_DIR }}" "{{ RABBITMQADMIN_URL }}" creates=rabbitmqadmin
```
[the remark this answers is not in the document]
Comment (pkulkark) [reply playbooks/roles/rabbitmq/tasks/main.yml]: The url is of a raw github file. For some reason, `get_url` module was failing so I went with the wget command instead. `creates=rabbitmqadmin` has to be inline (cf. [Another example](https://github.com/kamaln7/ansible-swapfile/blob/8eb0e18d90a7f25b6658fbf56bcba01b84a664fc/tasks/main.yml#L5)).

File: playbooks/roles/rabbitmq/defaults/main.yml [hunk 3]
```diff
@@ -29,3 +29,8 @@ RABBITMQ_SCRIPTS_DIR: /usr/local/sbin
 
 RABBITMQ_BACKUP_COMMAND: rabbitmq-backup
 RABBITMQ_RESTORE_COMMAND: rabbitmq-restore
+
+RABBITMQ_USER: set-me-please
+RABBITMQ_PASS: set-me-please
+RABBITMQ_VHOST: set-me-please
+RABBITMQADMIN_URL: https://raw.githubusercontent.com/rabbitmq/rabbitmq-management/v{{ RABBITMQ_VERSION[0:3]}}.x/bin/rabbitmqadmin
```
[the remark this answers is not in the document]
Comment (pkulkark) [reply playbooks/roles/rabbitmq/defaults/main.yml]: Thanks! I hadn't thought of that. Kept it harcoded with a comment for now.

File: playbooks/roles/rabbitmq/tasks/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (swalladge) [reply playbooks/roles/rabbitmq/tasks/main.yml]: Fair enough. Personally I don't like the inline args syntax as it's hard to differentiate between args that ansible uses and args that are passed to the command. That other example is especially confusing... 

According to the [docs](https://docs.ansible.com/ansible/latest/modules/command_module.html), we could do something like:

```
- name: download rabbitmqadmin executable
  command: wget -P "{{ RABBITMQ_SCRIPTS_DIR }}" "{{ RABBITMQADMIN_URL }}"
  args:
    creates: rabbitmqadmin
```

Actually, that will download the file to `{{ RABBITMQ_SCRIPTS_DIR }}/rabbitmqadmin`, while ansible will check for `rabbitmqadmin` which depends on the working directory. Perhaps it should either use the ansible `chdir` arg, or add the `{{ RABBITMQ_SCRIPTS_DIR }}` to `creates`?

[the remark this answers is not in the document]
Comment (pkulkark) [reply playbooks/roles/rabbitmq/tasks/main.yml]: Good point. I added `{{RABBITMQ_SCRIPTS_DIR }}` to `creates` and passed it as args. I also reran the playbook to verify everything works.

**Pull request merged**